skglm/penalties/separable.py

```diff
@@ -10,37 +10,49 @@
 class L1(BasePenalty):
     """L1 penalty."""
 
-    def __init__(self, alpha):
+    def __init__(self, alpha, positive=False):
         self.alpha = alpha
+        self.positive = positive
 
     def get_spec(self):
         spec = (
             ('alpha', float64),
+            ('positive', bool_),
         )
         return spec
 
     def params_to_dict(self):
-        return dict(alpha=self.alpha)
+        return dict(alpha=self.alpha, positive=self.positive)
 
     def value(self, w):
         """Compute L1 penalty value."""
         return self.alpha * np.sum(np.abs(w))
 
     def prox_1d(self, value, stepsize, j):
         """Compute proximal operator of the L1 penalty (soft-thresholding operator)."""
-        return ST(value, self.alpha * stepsize)
+        return ST(value, self.alpha * stepsize, self.positive)
 
     def subdiff_distance(self, w, grad, ws):
         """Compute distance of negative gradient to the subdifferential at w."""
         subdiff_dist = np.zeros_like(grad)
         for idx, j in enumerate(ws):
-            if w[j] == 0:
-                # distance of - grad_j to  [-alpha, alpha]
-                subdiff_dist[idx] = max(0, np.abs(grad[idx]) - self.alpha)
+            if self.positive:
```

Comment:
Suggestion that I think would enhance the readability of this part

```python
# case of positivity constraints on coefficients 
if positive:
    # loop over features to compute subdiff distance
    ...
    ...
    return subdiff distance

# usual case
...
....
...
```

WDTY? (@mathurinm any thoughts?)

Reply:
i find it redundant. I think it's already quite readable. @mathurinm  ?

Reply:
Apart from that, anything else? (any suggestions @mathurinm ?)